File: ApiSuvesaPos/Negocio/Logica/Presentaciones.vb
Namespace Logica
    Public Class Presentaciones


        Private db As Datos.Class.Presentaciones

        Sub New()
            Me.db = New Datos.Class.Presentaciones
        End Sub

        Public Function Buscar(porDescripcion As Boolean, Filtro As String) As List(Of Datos.Models.Presentacione)
            Return Me.db.Buscar(porDescripcion, Filtro)
        End Function

        Public Function Crear(presentacion As Datos.Models.Presentacione) As String
            Return Me.db.Crear(presentacion)
        End Function

        Public Function Editar(id As Integer, presentacion As Datos.Models.Presentacione) As String
            Dim res As String = Me.db.Editar(id, presentacion)
            If res = "0" Then
                Return "No existe el valor"
            Else
                Return res
            End If
        End Function

        Public Function Eliminar(id As Integer) As String
            Dim res As String = Me.db.Borrar(id)
            If res = "0" Then
                Return "No existe el valor"
            Else
                Return res
            End If
        End Function
    End Class

End Namespace

File: ApiSuvesaPos/Datos/Class/Presentaciones.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Datos.Models;
using Microsoft.EntityFrameworkCore;

namespace Datos.Class
{
  public  class Presentaciones
    {

        private SeePOSContext entities;

        public Presentaciones()
        {
            entities = new SeePOSContext();
        }

        public int Crear(Models.Presentacione presentacion) // crear carta apartir de objeto de creacion
        {
            try
            {
                entities.Presentaciones.Add(presentacion);
                return entities.SaveChanges();

            }
            catch (Exception ex)
            {
                throw ex;
            }

        }

        public List<Models.Presentacione> Buscar(bool porNombre, string filtro) //obtener lista CartasdeExoneracion 
        {
            try
            {
                List<Models.Presentacione> result;

                if (porNombre == true)
                {
                    //busca por la descripcion de la marca
                    var temp = from c in entities.Presentaciones
                                   // usar un like
                               select c;

                    result = temp.ToList<Models.Presentacione>();
                }
                else
                {
                    //busca por el id de la tabla
                    var temp = from c in entities.Presentaciones
                               where c.CodPres == Convert.ToInt32(filtro) //convertir el datos de string a long
                               select c;
                    result = temp.ToList<Models.Presentacione>();
                }

                if (result.Count > 0)
                {
                    return result;
                }
                else
                {
                    return result = null;
                }


            }
            catch (Exception ex)
            {
                throw ex;
            }
        }

        public int Editar(long id, Models.Presentacione presentaciones)
        {
            try
            {
                var p = entities.Presentaciones.Find(id);
                Models.Presentacione viejo = p;

                if (viejo != null)
                {
                    viejo.Presentaciones = presentaciones.Presentaciones;
                    viejo.Mh = presentaciones.Mh;

                    entities.Entry(viejo).State = EntityState.Modified;

                    return entities.SaveChanges();
                }
                else
                {
                    return 0;// no se encotro el registro solicitado.
                }

            }
            catch (Exception ex)
            {

                throw ex;
            }
        }

        public int Borrar(long id) // borrar Actividad apartir del id Actividad


        {
            try
            {
                var p = entities.Presentaciones.Find(id);
                entities.Remove(p);
                return entities.SaveChanges();

            }
            catch (Exception ex)
            {

                throw ex;
            }

        }


    }
}
